terraform/network.tf

```diff
@@ -0,0 +1,57 @@
+/**
+* Networking to support private clusters. This config is only deployed when the
+* enable_private_cluster variable is set to true.
+* Adapted from 2i2c: https://github.com/2i2c-org/infrastructure/pull/538/files
+*/
+
+data "google_compute_network" "default_network" {
+  name    = "default"
+  project = var.project_name
+}
+
+data "google_compute_subnetwork" "default_subnetwork" {
+  name    = "default"
+  project = var.project_name
+  region  = var.cluster_region
+}
+
+resource "google_compute_firewall" "iap_ssh_ingress" {
+  count = var.enable_private_cluster ? 1 : 0
+
+  name    = "allow-ssh"
+  project = var.project_name
+  network = data.google_compute_network.default_network.name
+
+  allow {
+    protocol = "tcp"
+    ports    = ["22"]
+  }
+
+  // This range contains all IP addresses that IAP uses for TCP forwarding.
+  // https://cloud.google.com/iap/docs/using-tcp-forwarding
+  source_ranges = ["35.235.240.0/20"]
```

Comment:
This should be fed in as a variable with a default, rather than hardcoded.

Reply:
Quick one, what is IAP?

Reply:
and we should still do this. No idea about IAP, probably have to ask Yuvi or Sarah.

Reply:
*Note: probably worth adding to the docs how the interaction with IAP works for this usecase
Also, now on the fence RE: putting this in a variable after pondering it some more. 

If this were to be used in another terraform project as a module, it may be more worthwhile, but I think that is unlikely in this case. The only time I can see this being changed is to add other whitelist IPs for on-prem access from static IPs (Non-Columbia usecases)

Reply:
For the benefit of others having googled it: IAP is Google's Identity Aware Proxy :)